### ISSUE TITLE
Validation for available commands.

### ISSUE THREAD

**Issue by Souvikns** [asyncapi/cli#70]
#### Reason/Context
Just what @derberg said https://github.com/asyncapi/cli/pull/59#issuecomment-915152405 it would be nice to have a validation service that prompts when the user passes any command or flag which is not intended and giving proper help messages and helping the user.

**Comment by derberg** [comment]
yup, still needed, we use https://www.npmjs.com/package/levenshtein-edit-distance already in generator -> https://github.com/asyncapi/generator/blob/master/lib/templateConfigValidator.js#L67

it is pretty nifty and should not be difficult to add I think

**Comment by ron-debajyoti** [comment]
Hey, can I take this up?

**Comment by derberg** [comment]
@ron-debajyoti definitely 👍🏼

**Comment by ron-debajyoti** [comment]
Hi @Souvikns ,  
I was exploring to fix the issue, but I can't really figure out where do I catch the mismatch of the commands. 

For example : 
- when I try to run `asyncapi validate wrongCommand` or a command where I use one of the supported commands but a wrong input parameter, then the `catch` function in `src/base.ts` catches it, but 
- when I try to run `asyncapi invalidCommand`, I don't see the above `catch` function in `src/base.ts` catching it. I think `oclif` directly handles there isn't such a class defined under `src/commands`. How do I handle such cases?

Please correct me if I got something wrong.

**Comment by Souvikns** [comment]
Yeah, you are right if there isn't a class in the commands folder then oclif handles it. You can refer to [this](https://oclif.io/docs/error_handling). Also, check [this](https://oclif.io/docs/plugins) out for this issue, there are some premade plugins that would help us achieve this.

**Comment by ron-debajyoti** [comment]
Hey @Souvikns , sorry for not responding to this earlier.

I looked into the plugins and figured using `@oclif/plugin-not-found` would solve the problem. I tried installing it and also `@oclif/plugin-help`, but I'm getting plugin module errors during executing. Trying `prepack` also showed the import errors.

The error looks like this : 
![Screenshot from 2022-02-21 21-35-40](https://user-images.githubusercontent.com/22571664/154990877-0e1de2f5-a972-417f-9459-457891da7fee.png)



I suspect this might be due to  `@oclif/plugin-help` dependent on `@oclif/core` and we are using `@oclif/config`, Seems like have to make our own plugins instead of using existing ones  but I would like to have you take a look into it as well.

**Comment by Souvikns** [comment]
There is a bug, CLI is breaking in production, it runs fine but shows a warning, you can look into https://github.com/asyncapi/cli/issues/192. We are migrating to a new version of oclif https://github.com/asyncapi/cli/pull/203. Can you merge this branch and try once. 

Let me know if this problem is resolved.

**Comment by ron-debajyoti** [comment]
> Let me know if this problem is resolved.
Yep, it worked! 

So since you're waiting for that PR to get merged, do I open a PR too for this issue?

**Comment by Souvikns** [comment]
Yeah @ron-debajyoti you can go ahead and open a PR.
